api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py

```diff
@@ -636,7 +636,7 @@ def _convert_prompt_message_to_dict(self, message: PromptMessage) -> dict:
             #     "tool_call_id": message.tool_call_id
             # }
             message_dict = {
-                "role": "function",
+                "role": "tool" if credentials and credentials.get('function_calling_type', 'no_call') == 'tool_call' else "function",
```

Comment:
On line 687, this method is also called. I think you need to pass the credentials there as well.

Reply:
Thank you for catching that oversight. I have addressed your comment by updating line 687 to include the necessary credentials.